RdpCredProv/RdpProvider.cpp

```
@@ -205,36 +238,63 @@ STDMETHODIMP RdpProvider::SetSerialization(const CREDENTIAL_PROVIDER_CREDENTIAL_
 
 HRESULT RdpProvider::Advise(__in ICredentialProviderEvents* pcpe, UINT_PTR upAdviseContext)
 {
-	UNREFERENCED_PARAMETER(pcpe);
-	UNREFERENCED_PARAMETER(upAdviseContext);
+	g_log.Write("DEBUG: Advise called - setting up event callbacks");
+	g_log.Write("DEBUG: ICredentialProviderEvents pointer: %p, AdviseContext: %llu", pcpe, upAdviseContext);
+
+	if (_pCredentialProviderEvents)
+	{
+		g_log.Write("DEBUG: Releasing existing credential provider events");
+		_pCredentialProviderEvents->Release();
+	}
+
+	_pCredentialProviderEvents = pcpe;
+	_pCredentialProviderEvents->AddRef();
+	_upAdviseContext = upAdviseContext;
 
-	log.Write("RdpProvider::Advise");
+	g_log.Write("DEBUG: Event callbacks set up successfully");
 
-	return E_NOTIMPL;
+	// Try to connect to desktop client (non-blocking)
+	g_log.Write("DEBUG: Attempting initial connection to desktop client");
+	HRESULT hrConnect = _ConnectToDesktopClient();
```

the error is ignored